### PR TITLE
Detect and use Alire metadata found in pinned directory

### PR DESCRIPTION
**Pull request by mosteo**
Goes on top of #449.

PR #447 introduces the ability to pin a folder that needs not be an Alire crate. This allows the use of Ada code entirely out of the Alire ecosystem to fulfill a dependency. The shortcoming is that any dependencies introduced by such code will remain unknown and have to be manually dealt with. Also, if the target folder was, indeed, an Alire crate, its information is ignored.

This is an incremental feature in which, if the target folder contains Alire metadata, it is loaded and used to 

- Verify the crate we are pinning matches its dependency in name
- Add its dependencies during resolution
- Add its project paths during building
- Optionally, autodetect the crate for a shorthand `alr with --url` form.

In short, the pinned Alire crate is used normally as if it had been retrieved as a regular dependency, except for the version, which is ignored (the dependency will always be fulfilled, as if it were a "raw" pinned folder).

### CODE REVIEW

File: doc/user-changes.md [hunk 1]
```diff
@@ -4,6 +4,19 @@ This document is a development diary summarizing changes in `alr` that notably
 affect the user experience. It is intended as a one-stop point for users to
 stay on top of `alr` new features.
 
+## Use crate metadata when pinning to a directory
```
Comment (Fabien-Chouteau) [inline doc/user-changes.md]: Awesome I was about to suggest that when reading the main PR :)

**Comment by Fabien-Chouteau** [comment]
>     * Add its dependencies during resolution

If there are more than one release in the crate, which one is going to be used for that?

**Review by Fabien-Chouteau — APPROVED** [review]

**Comment by mosteo** [comment]
> If there are more than one release in the crate, which one is going to be used for that?

This should cause an error, because we expect the local `alire/crate.toml` to contain only the checked out release metadata. When you `alr get` a release that's what's stored. For a never-published crate, `init` also only creates one release.

Actually this ties in a bit with a follow-up discussion I want to have about the maintenance/publishing workflow, because I think we should move to making `alire/crate.toml` manually edited only and put under version control. I'm organizing my ideas and will open an issue soon before doing anything.
